tools/write_release_and_log.py

```diff
@@ -0,0 +1,145 @@
+"""
+Standalone script for writing release doc and logs
+-------------------------------------------------
+Cmd -> python write_release_and_log.py <LOG_START> <LOG_END>
+Example -> python write_release_and_log.py v1.6.0 v1.8.0
+"""
+
+import os
+import sys
+import glob
+import subprocess
+from hashlib import md5
+from hashlib import sha256
+from pathlib import Path
+
+sys.path.insert(0, os.path.dirname(__file__))
+sys.path.insert(1, os.path.join(os.path.dirname(__file__), 'tools'))
+
+
+try:
+    version_utils = __import__("version_utils")
+    FULLVERSION = version_utils.VERSION
+    # This is duplicated from setup.py
+    if os.path.exists('.git'):
+        GIT_REVISION, _ = version_utils.git_version(
+            os.path.join(os.path.dirname(__file__), '..'))
+    else:
+        GIT_REVISION = "Unknown"
+
+    if not version_utils.ISRELEASED:
+        if GIT_REVISION == "Unknown":
+            FULLVERSION += '.dev0+Unknown'
+        else:
+            FULLVERSION += '.dev0+' + GIT_REVISION[:7]
+finally:
+    sys.path.pop(1)
+    sys.path.pop(0)
+
+try:
+    # Ensure sensible file permissions
+    os.umask(0o022)
+except AttributeError:
+    # No umask on non-posix
+    pass
+
+
+def get_latest_release_doc(path):
+    """
+    Method to Pick the latest file from 'doc/release'
+    """
+    files_path = os.path.join(path, '*')
+    files = sorted(
+        glob.iglob(files_path), key=os.path.getmtime, reverse=True)
```

Comment:
This sorting is incorrect. It should be sorting by filename, not by file timestamp. When trying `python tools/write_release_and_log.py v1.8.0 HEAD` I saw the `1.8.0` release notes being picked up, which `1.9.0` is the highest number. I suggest splitting the file name on `.`, taking the first two parts of the returned list, and sorting first on `MAJOR` then on `MINOR` version (convert strings to integers first before sorting - you can check it's correct by adding `doc/release/1.10.0-notes.rst` and verifying it gets picked up as higher than `1.9.0`)

Reply:
Tweaked the logic based on the above suggestion and also performed a small test.

Reply:
Works like a charm now.